presto-main/src/main/java/com/facebook/presto/operator/aggregation/state/KeyValuePairStateSerializer.java

```diff
@@ -44,7 +51,7 @@ public void serialize(KeyValuePairsState state, BlockBuilder out)
     public void deserialize(Block block, int index, KeyValuePairsState state)
     {
         if (!block.isNull(index)) {
-            state.set(new KeyValuePairs(VARBINARY.getSlice(block, index), state.getKeyType(), state.getValueType()));
```

Comment:
now that the serializer doesn't use `state.getKeyType` and `getValueType` can you remove them from the interface and replace the usage in `MapAggregation` with `bindTo`?